pallets/asset/src/identifier.rs

```diff
@@ -0,0 +1,176 @@
+use codec::{Decode, Encode};
+use core::convert::TryInto;
+use sp_std::vec::Vec;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub enum Identifier {
+    CUSIP([u8; 9]),
+    CINS([u8; 9]),
+    ISIN([u8; 12]),
+    LEI([u8; 20]),
+    EMPTY,
+}
+
+impl Default for Identifier {
+    fn default() -> Self {
+        Identifier::EMPTY
+    }
+}
+
+impl Identifier {
+    pub fn cusip(bytes: [u8; 9]) -> Option<Identifier> {
+        (cusip_checksum(&bytes[..8]) == bytes[8] - b'0').then_some(Identifier::CUSIP(bytes))
+    }
+
+    pub fn cins(bytes: [u8; 9]) -> Option<Identifier> {
+        (cusip_checksum(&bytes[..8]) == bytes[8] - b'0').then_some(Identifier::CINS(bytes))
```

Comment:
Does cins and cusip have identical algorithms?

Reply:
yes...

Reply:
@CJP10 can we remove the duplicated code?

Reply:
Originally, I did more than just validate the checksum but in it's current state it's duplicated.